### DISCUSSION TITLE
Wall jumping movements causing disconnect.

### DISCUSSION THREAD

**Discussion by AngelWolf** [nefarius/DsHidMini#98] [Q&A] [answered]
So I was playing Ratchet and Clank last night and I came to a part of the game where you need to wall jump up a vertical shaft by moving the left stick side to side in sync with the jump button.  After a few attempts the controller just disconnected causing me to fall back to the beginning.  I thought it might be some odd emulation glitch as it only seemed to happen there.  Anyways, this evening I decided to try an experiment.  I connected my genuine DS3 controller via bluetooth and without even starting up the emulator, I made the movements as if I were playing a game and the controller disconnected.  I've repeated this test multiple times with the same result.  I can't seem to enable logging in the control program.  I tick the box and restart the program, but the box is unticked when I start it up again and there are no logs.  Is there anything I can do to stop the controller from disconnecting?

**Comment by nefarius** [comment]
The logs of DSHMC only cover the UI tool anyway, to get to see what happens in the Bluetooth driver [run a trace log](https://vigem.org/projects/BthPS3/Debugging-the-drivers/) for the entire session and capture the disconnect. Compress and attach the generated ETL file here. Other than that if you don't accidentally hit the quick disconnect combo (see README) I am not aware of any logic that would cause a disconnect. Also look in the Windows Event Log for user mode driver crash entries.

Cheers

**Comment by AngelWolf** [comment]
[BthPS3.zip](https://github.com/ViGEm/DsHidMini/files/6681544/BthPS3.zip)
Here's the log.  I'm in SDF mode, rocking the left control stick left and right while pressing the X button in time.  The problem is easy to replicate if you do a 4 count, so left + X, pause, right + X, pause

**Comment by AngelWolf** [comment]
The disconnect seems to be happening without the bluetooth dongle even connected.  It happens with the controller connected with USB cable as well.  Kept getting random disconnects while playing Ratchet and Clank while doing random stuff.

**Reply by nefarius** [reply]
Not a good sign, what's your battery status on wireless? Maybe it's aged and can't even hold enough charge, even during USB play. Software-wise I have no clue what could be going on.

**Reply by AngelWolf** [reply]
Battery seems fine, but it's also 5 years old, so I'm guessing it's having voltage drops that are causing disconnects.  Just finished the evening with a second controller that had no problems whatsoever, so I'm hoping my primary just needs a new battery.  Thanks for the kick ass program nefarious!

**Comment by AngelWolf** [comment, marked as answer]
Just finished putting in a new battery and playing in the section that made the controller spaz out.  Everything working like normal.  Thanks again nefarius!
